crates/nu-cli/src/commands/sleep.rs

```diff
@@ -36,7 +45,26 @@ impl WholeStreamCommand for Sleep {
         args: CommandArgs,
         registry: &CommandRegistry,
     ) -> Result<OutputStream, ShellError> {
-        sleep(args, registry).await
+        let registry = registry.clone();
+        let ctrl_c = args.ctrl_c().clone();
+
+        let (SleepArgs { duration, rest }, input) = args.process(&registry).await?;
+
+        let total_dur = Duration::from_nanos(duration.item)
+            + rest
+                .iter()
+                .map(|val| Duration::from_nanos(val.item))
+                .sum::<Duration>();
+
+        SleepFuture::new(total_dur, ctrl_c).await;
+        // this is necessary because the following 2 commands gave different results:
+        // `echo | sleep 1sec` - nothing
+        // `sleep 1sec`        - table with 0 elements
+        if input.is_empty() {
+            Ok(OutputStream::empty())
+        } else {
+            Ok(OutputStream::from_input(input))
+        }
```

Comment:
This really feels like something `OutputStream::from_input` should do. What do you think about moving the `is_empty` check to `from_input` instead?

Reply:
I can put it in `From<InputStream> for OutputStream`, `OutputStream::from_input` has more relaxed trait boundaries so I can't do that there. This check in `From<InputStream>` would create a difference between those two calls, and may cause different results from different commands.

`InputStream::is_empty` is special-cased, it is only set as empty when explicitly created that way.

An alternative I can see would be making `autoview` output nothing both when the stream is explicitly empty and when there are 0 values, instead of outputting an empty table (like it does now).

Reply:
Oh, you're totally right. I completely missed that `from_input` is generic on the `input` parameter.

> I can put it in `From<InputStream> for OutputStream`, `OutputStream::from_input` has more relaxed trait boundaries so I can't do that there. This check in `From<InputStream>` would create a difference between those two calls, and may cause different results from different commands.

Part of me thinks anything else would be an incorrect transformation, but let's not worry about that here. I'm digressing from the PR 😅

Reply:
I found that `autoview` does not match on Primitive::Nothing, I think it's better to print nothing instead of an empty table.

Then, I could remove this if and not create a difference between `.into()` and `OutputStream::from_input()`.